plugins/reporters/opossum/src/main/kotlin/OpossumReporter.kt

```diff
@@ -500,53 +573,31 @@ class OpossumReporter(
         }
     }
 
+    @Serializable
+    internal data class OpossumInputMetadata(
+        val projectId: String = "0",
+        val fileCreationDate: String = LocalDateTime.now().toString()
+    )
+
     private fun writeReport(outputFile: File, opossumInput: OpossumInput) {
         val jsonFile = createOrtTempDir().resolve("input.json")
-        JsonMapper().setSerializationInclusion(Include.NON_NULL).writeValue(jsonFile, opossumInput.toJson())
+        val json = Json {
+            explicitNulls = false
+            encodeDefaults = true
+        }
+        jsonFile.writeText(json.encodeToString(OpossumInput.serializer(), opossumInput))
+
         jsonFile.packZip(outputFile)
         jsonFile.delete()
     }
 
-    internal fun generateOpossumInput(input: ReporterInput, maxDepth: Int = Int.MAX_VALUE): OpossumInput {
-        val opossumInput = OpossumInput()
-
-        opossumInput.addBaseUrl("/", input.ortResult.repository.vcs)
-
-        SpdxLicense.entries.forEach {
-            val licenseText = input.licenseTextProvider.getLicenseText(it.id)
-            opossumInput.frequentLicenses += OpossumFrequentLicense(it.id, it.fullName, licenseText)
-        }
-
-        input.ortResult.getProjects().forEach { project ->
-            opossumInput.addProject(project, input.ortResult)
-        }
-
-        if (input.ortResult.getExcludes().scopes.isEmpty()) {
-            opossumInput.addPackagesThatAreRootless(input.ortResult.getPackages())
-        }
-
-        input.ortResult.analyzer?.result?.issues.orEmpty().forEach { (id, issues) ->
-            issues.forEach { issue ->
-                val source = opossumInput.addExternalAttributionSource(
-                    key = "ORT-Analyzer-Issues",
-                    name = "ORT-Analyzer Issues",
-                    priority = ISSUE_PRIORITY
-                )
-
-                opossumInput.addIssue(issue, id, source)
-            }
-        }
-
-        input.ortResult.getScanResults().forEach { (id, results) ->
-            opossumInput.addScannerResults(id, results, maxDepth)
-        }
-
-        return opossumInput
+    internal fun createOpossumInput(input: ReporterInput, maxDepth: Int = Int.MAX_VALUE): OpossumInput {
+        return OpossumInputCreator().create(input, maxDepth)
```

Comment:
This could as well be a function expression.

Reply:
Tbh, I do not like them too much... But I can change it if you prefer.

Reply:
We have a (non-enforced) convention to use expression functions if the body only consist of a `return` statement. But this could also be cleaned up later.

Reply:
Okay, that's fine. I changed it already.